ot/optim.py

```diff
@@ -69,7 +69,7 @@ def phi(alpha1):
     alpha, phi1 = scalar_search_armijo(
         phi, phi0, derphi0, c1=c1, alpha0=alpha0)
 
-    return min(1, alpha), fc[0], phi1
+    return min(1, alpha) if alpha is not None else 1, fc[0], phi1
```

Comment:
could you put the test in another line with a small comment about what is necessary?

It is a bit hard to follow in the return

Reply:
Also I guess you have an example where the original function failed. Could you do a simple no regression test it test_optim.py to check that this problem never re-appears again? It feels as if this function always needs tuning ;)

Reply:
Hello @panpan2 could you please take into account my comment so that we can merge your PR?

Reply:
Updated! :)

Reply:
@rflamary any other changes needed to merge this?